bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/TaskBar.java

```diff
@@ -166,14 +167,14 @@ IShellLink createShellLink (MenuItem item) {
 				ImageData data;
 				if (item.hBitmap != 0) {
 					Image image2 = Image.win32_new (display, SWT.BITMAP, item.hBitmap);
-					data = image2.getImageData (DPIUtil.getDeviceZoom ());
+					data = image2.getImageData (getZoom());
```

Comment:
Can't the zoom of the `MenuItem` passed to the method be used instead of using the zoom of the `TaskBar`, which is simply the one of the primary monitor? I would expect this to usually use the wrong scale value.

Reply:
I thought `getZoom()` gets the current zoom (`nativeZoom` located in `Widget`), isn't that so? 

The way I understood it is that every time an event of the type `SWT.ZoomChanged` is triggered, this ends up calling `org.eclipse.swt.widgets.Widget.handleDPIChange(Widget, int, float)`, which sets `nativeZoom`.

On the other hand, I don't see any special handler registration in `TaskBar` as other widgets have (_e.g._ `ExpandBar`).

Reply:
Hmm, yes, I think in this context using the zoom from menu item makes probably one sense. I will have a second look at it

Reply:
I am not sure if (and under which conditions) `TaskBar` receives any zoom change events from the OS. Maybe this happens when the primary monitor or its DPI settings change? Anyway, if no zoom change happens, the `TaskBar` is initialized with a `nativeZoom` given by `display.getDeviceZoom()` (see the other change in this PR). That one always originates from the primary monitor.